### PR TITLE
TASK: Move imageOptions to configuration

### LINKED ISSUE #17: Imagesizeoptions can not be modified
The imagesize options are hardcoded and can not be modified

### PR DESCRIPTION
**Pull request by sorenmalling**
Resolves #17

### CODE REVIEW

**Comment by sorenmalling** [comment]
Talking to @kdambekalns about confirmation of the "concept".

Alternative: Remove the "height" property from the current imageOptions, to have the preview images in the media browser not cropped, so the editor can see if they are vertical or horizontal

**Review by kdambekalns — COMMENTED** [review]
Thanks! Looks good, I'll test this ASAP.

**Comment by sorenmalling** [comment]
@kdambekalns I will suggest that we use keys for the different sizes - that way, it's easier to override the imageOptions for a specific case. Do you agree on that thought?

**Comment by sorenmalling** [comment]
@kdambekalns Named configurations with preset fallback is ready for review

**Comment by kitsunet** [comment]
Mmm, I feel this should be a major version due to the fact the configuration has changed? @sorenmalling @kdambekalns

**Comment by kdambekalns** [comment]
> Mmm, I feel this should be a major version due to the fact the configuration has changed?

In fact the configuration does what was hardcoded before – except the cropping for the thumbnails. So I'd say this can go into a minor…

File: Classes/AssetSource/PixxioAssetSource.php [hunk 1]
```diff
@@ -145,6 +150,12 @@ public function __construct(string $assetSourceIdentifier, array $assetSourceOpt
                     }
                     $this->apiClientOptions = $optionValue;
                     break;
+                case 'imageOptions':
+                    if (!is_array($optionValue)) {
+                        throw new \InvalidArgumentException(sprintf('Invalid image options specified for Pixx.io asset source %s', $assetSourceIdentifier), 1591605348);
```
Comment (kdambekalns) [inline Classes/AssetSource/PixxioAssetSource.php:155]: I'll give that a unique exception key…

**Pull request merged**